File: resource-api/app/models/share.py
from pydantic import BaseModel


class ShareCreate(BaseModel):
    name: str
    url: str
    avatar_url: str = ""
    description: str = ""
    sort_order: int = 0


class ShareUpdate(BaseModel):
    name: str | None = None
    url: str | None = None
    avatar_url: str | None = None
    description: str | None = None
    sort_order: int | None = None


class ShareOut(BaseModel):
    id: int
    name: str
    url: str
    avatar_url: str
    description: str
    sort_order: int
    created_at: str | None = None
    updated_at: str | None = None

File: resource-admin/src/api/share.ts
import http from './index'
import type { ApiResponse } from '../types'

export interface Share {
  id: number
  name: string
  url: string
  avatar_url: string
  description: string
  sort_order: number
  created_at: string | null
  updated_at: string | null
}

export interface ShareForm {
  name: string
  url: string
  avatar_url?: string
  description?: string
  sort_order?: number
}

export const shareApi = {
  list: () =>
    http.get<ApiResponse<Share[]>>('/admin/shares'),

  create: (data: ShareForm) =>
    http.post<ApiResponse<Share>>('/admin/shares', data),

  update: (id: number, data: Partial<ShareForm>) =>
    http.put<ApiResponse<Share>>(`/admin/shares/${id}`, data),

  delete: (id: number) =>
    http.delete<ApiResponse<null>>(`/admin/shares/${id}`),
}
